### PR TITLE
Rewritten hostname extraction from resolver URL

### LINKED ISSUE #160: Hostname is not resolved if it contains a port at the end
[`hostname_from_uri`](https://github.com/aarond10/https_dns_proxy/blob/d03e11572562f008f68df217a7378628f1bb7b79/src/main.c#L43) fails if the address is a hostname (that should be resolved) followed by a port.

### PR DESCRIPTION
**Pull request by baranyaib90**
Hi, this change fixes #160 
Beside that I would recommend to close all the current issues. I think all of them are answered or we have nothing to do with them.

### CODE REVIEW

**Comment by baranyaib90** [comment]
Nope, do not merge it yet 😅
I have forgot to test with IP address in the URL.

**Comment by baranyaib90** [comment]
Now it is fixed, should be working properly.
Piggyback: some very minor logging fix.

File: src/main.c [hunk 1]
```diff
@@ -36,34 +36,33 @@ typedef struct {
...
-
-  // If using basic authentication in URL, chop off prefix.
-  char *tmp = strchr(uri, '@');
-  if (tmp) {
-    tmp++;
-    if (tmp < end) {
-      uri = tmp;
+static int is_ipv4_address(char *str) {
+    struct in6_addr addr;
+    return inet_pton(AF_INET, str, &addr) == 1;
```
Comment (aarond10) [inline src/main.c:41]: Much nicer. Thanks!

File: src/main.c [hunk 2]
```diff
@@ -36,34 +36,33 @@ typedef struct {
...
+}
+
+static int hostname_from_url(const char* url_in,
```
Comment (aarond10) [inline src/main.c:44]: Nit: May be worth keeping the return code comment (non-zero == success, zero = failure).

**Review by aarond10 — APPROVED** [review]

File: src/main.c (hunk 1 above)
Comment (baranyaib90) [reply src/main.c:41]: I hoped so. You are welcome!

File: src/main.c (hunk 2 above)
Comment (baranyaib90) [reply src/main.c:44]: Actually I have left a very minimal trace of that comment by indicating which value means success. I thought it is enough.
`res = 1; // success`

**Pull request merged**